src/expression/comparison/index.ts

```diff
@@ -72,6 +83,24 @@ export abstract class Comparison implements Evaluable {
     if (operator === undefined) {
       throw new Error(`missing operator ${this.operatorSymbol.toString()}`)
     }
-    return [operator, this.left.serialize(options), this.right.serialize(options)]
+    return [
+      operator,
+      this.left.serialize(options),
+      this.right.serialize(options)
+    ]
+  }
+
+  protected parseable (value: Result): number | undefined {
```

Comment:
This more likely: `toNumber` and it should not return `undefined` rather the value it self in the default case. 

i.e. 

protected toNumber (value: Result): number | value {
   // ... parsing
   
   return value
}

Reply:
Add a comment:

`Cast or convert a value to number if possible, otherwise return the input value untouched` or something like that

Reply:
9b2e2c0